BlazarUI/app/scripts/components/shared/ModuleModal.jsx

```diff
@@ -0,0 +1,83 @@
+import React, {Component, PropTypes} from 'react';
+import Immutable from 'immutable';
+import Modal from 'react-bootstrap/lib/Modal';
+import Button from 'react-bootstrap/lib/Button';
+import Checkbox from './Checkbox.jsx';
+import $ from 'jquery';
+
+class ModuleModal extends Component {
+
+  renderModule(moduleObject) {
+    let moduleId = moduleObject.get('id');
+    let moduleName = moduleObject.get('name');
+
+    return (
+      <div>
+        <Checkbox
+          label={moduleName}
+          name='module-checkbox'
+          value={moduleId}
+          checked={true} />
+        <br />
+      </div>
+    );
+  }
+
+  renderDownstreamToggle() {
+    return (
+      <Checkbox
+        label='Build Downstream Modules'
+        name='downstream-checkbox' />
+    );
+  }
+
+  renderModuleList() {
+    return (
+      <div className='module-modal--container'>
```

Comment:
Thanks for the BEM reminder

 to this and your other comments re: css and styling, I intend to overhaul the entire modal format anyway. Mostly likely all of it will be bootstrap anyway